jlm/llvm/ir/operators/IntegerOperations.cpp

```diff
@@ -0,0 +1,572 @@
+/*
+ * Copyright 2025 Nico Reißmann <nico.reissmann@gmail.com>
+ * See COPYING for terms of redistribution.
+ */
+
+#include <jlm/llvm/ir/operators/IntegerOperations.hpp>
+
+namespace jlm::llvm
+{
+
+IntegerBinaryOperation::~IntegerBinaryOperation() = default;
+
+IntegerAddOperation::~IntegerAddOperation() noexcept = default;
+
+bool
+IntegerAddOperation::operator==(const Operation & other) const noexcept
+{
+  const auto addOperation = dynamic_cast<const IntegerAddOperation *>(&other);
+  return addOperation && addOperation->result(0) == result(0);
```

Comment:
Could all the references to `result(0)` use the `Type()` method instead?

Reply:
Fixed.